### PR TITLE
ZH-351 - Improved automatic linting and updated existing SOPs

### PR DESCRIPTION
**Pull request by M-casado**
## Summary
Improved the linting scripts in order for the automatic checks to be more informative and check further that SOPs comply with the style guide.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New content (non-breaking change which adds new content)
- [x] Modified content (non-breaking change which modifies existing content)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Motivation and Context
Checking the format of SOPs is a tiring task if done manually. The automatic checks will serve as a quick and informative check of whether SOPs conform with the expected format or not.

### References
[ZenHub #351](https://app.zenhub.com/workspaces/t43-gdi-sops-662a0b0fe38b0405fb600d67/issues/zh/351)

## Changes Introduced
- Added new rules to the linting script
- Updated documentation where needed
- Updated the three existing SOPs to comply with these linting rules
- Updated the Charter's glossary

## Review
NA

## Additional Notes
NA

## Checklist:
**General Compliance:**
- [x] My changes follow the code style of this project ([GDI SOP Style Guide](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_style-guide.md)) and the file naming conventions of the [file accessioning proposal](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs//GDI-SOP_sop-accessioning.md).
- [x] I have verified that all new updated content is accessible, including checking that all external references are readable (i.e., no broken links). These may include references to external resources that should be resolvable, and internal references among SOPs.
- [x] I have properly added this PR's changes to the repository [CHANGELOG.md](../CHANGELOG.md).

Only applicable if the PR includes new, or changes to, GDI SOPs (i.e., documents at ``sops/``):
- [x] My SOP-related changes adhere to the [Generic SOP Template](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_sop-template.md), including format and required fields.
- [x] I have consulted the [Charter](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_charter.md), [ISM](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_information-service-management.md), and [ORR](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_organisational-roles-and-responsibilities.md) documents to ensure compliance. 
- [x] I am complying with the established procedure for SOP creations and modifications, including respecting review phases and notifying needed contributors for reviews.

### CODE REVIEW

**Comment by M-casado** [comment]
For the reviewers: I wouldn't recommend focusing much on the code itself, but overall on the changes on the three SOPs, and whether they make sense or not.

**Comment by M-casado** [comment]
I would recommend any reviewer to start checking what the CHANGELOG changes are, which contains summary of each file that changed, and what changed.

File: docs/GDI-SOP_charter.md [hunk 1]
```diff
@@ -20,54 +20,72 @@
...
+| WP               | Work Package                                                            |
+| ZH               | ZenHub                                                                  |
+
+| **Term**         | **Definition**                                                         |
+|------------------|-------------------------------------------------------------------------|
+| CHANGELOG        | Log or record of all notable changes made to a project                 |
+| FitSM            | Standards for lightweight IT Services Management; developed initially through The FedSM Project, funded by the European Commission, now maintained by ITEMO (<https://www.fitsm.eu/>) |
+| Permit Authority | National body responsible for data access decisions                    |
+| README           | File that provides information about a project or directory            |
+| YYYYMMDD         | Date format in year-month-day                                          |
```
Comment (elisavettorstensson) [inline docs/GDI-SOP_charter.md:87]: This is a minor detail, but if you want to use ISO 8601 for the dates, then the format should be like this: YYYY-MM-DD. But we do not need to exaggerate so much! :-)

File: sops/european-level/GDI-SOP0003_1+mg-dac-recommendation-approval.md [hunk 2]
```diff
@@ -33,12 +34,25 @@
 Find GDI SOPs common Glossary at the [**charter document**](https://github.com/GenomicDataInfrastructure/standard-operating-procedures/blob/main/docs/GDI-SOP_charter.md).
 
 For this SOP in particular, the source of truth for most definitions and terms is the [1+MG Data Access Governance for Research](https://docs.google.com/document/d/19HVf7SP2R_fCI-KugVStZR4yEAkV6vhvSbVd6a2ZviU).
+
 | Abbreviation | Description                                                                                               |
```
Comment (elisavettorstensson) [inline sops/european-level/GDI-SOP0003_1+mg-dac-recommendation-approval.md:38]: In this list, it seems that you do not follow the alphabetical order. Is there any reason for that?

File: tests/GDI-SOP0000_sop-template-for-linting.md [hunk 3]
```diff
@@ -34,11 +36,16 @@ Find GDI SOPs common Glossary at the [**charter document**](https://github.com/G
 
 | Abbreviation | Description     |
```
Comment (elisavettorstensson) [inline tests/GDI-SOP0000_sop-template-for-linting.md:37]: The same here, no alphabetical order.

**Review by elisavettorstensson — COMMENTED** [review]
I’ve gone through the sections where you made changes, reviewed the changelog, and read through what you’ve documented there. I added a few minor comments but didn’t have anything specific to note about the Python code. I’m not entirely sure if there’s anything more I should have done for the review, so I don’t feel 100% confident that my review covers everything as it should. Let me know if I should approach it differently! :-)

File: docs/GDI-SOP_charter.md (hunk 1 above)
Comment (M-casado) [reply docs/GDI-SOP_charter.md:87]: To be honest I have no strong feelings about this at all. The ``YYYYMMDD`` references the bit in ``GDI-SOP0007``:

> Name the new file following the format of ``< YYYYMMDD > - GDI-SOP_draft-< SOP title >`` (e.g., 20240702 - GDI-SOP_draft-SOP Template creation).

It's normally the way I have named filenames to keep them ordered by date, but there's no reason not to change to YYYY-MM-DD 🤔

File: sops/european-level/GDI-SOP0003_1+mg-dac-recommendation-approval.md (hunk 2 above)
Comment (M-casado) [reply sops/european-level/GDI-SOP0003_1+mg-dac-recommendation-approval.md:38]: The reason is likely that I didn't pay attention 😅 
Thanks for picking it up! I'll fix the glossaries. I tend to order them, and then add a new one and forget to reorder.

File: tests/GDI-SOP0000_sop-template-for-linting.md (hunk 3 above)
Comment (M-casado) [reply tests/GDI-SOP0000_sop-template-for-linting.md]: Done

**Comment by M-casado** [comment]
> I’m not entirely sure if there’s anything more I should have done for the review

@elisavettorstensson - It's perfectly fine, if anyone is familiar with coding in Python, then they're welcome to review the code parts. Otherwise, reviewing the changes and the CHANGELOG is enough, in my opinion.

File: docs/GDI-SOP_charter.md (hunk 1 above)
Comment (elisavettorstensson) [reply docs/GDI-SOP_charter.md:87]: As I said ... it is a minor detail! Leave it as it is!! :-)

**Comment by M-casado** [comment]
Mentioned at today's meeting - Briefly explained the goal of the code, albeit the code itself is not reviewed yet. Being that it's not user-facing and its role is to lint, and given our team's code familiarity, I'll merge it as is.

**Pull request merged**